src/app/core/public/api/Circuit.ts

```diff
@@ -82,7 +87,10 @@ export interface Circuit {
     attachCanvas(canvas: HTMLCanvasElement): () => void;
     detachCanvas(): void;
 
-    addRenderCallback(cb: () => void): void;
+    forceRedraw(): void;
+
+    // TODO[](leon) - Need to make a public-facing RenderHelper/RenderOptions
+    addRenderCallback(cb: (data: { renderer: RenderHelper, options: RenderOptions, circuit: Circuit }) => void): void;
```

Comment:
Also what do you think about this @KevinMackenzie ?

Reply:
It's a little gross as-is, but it seems like a reasonable compromise until we separate the existing `CircuitView` from the rendering itself, as mentioned in the IC's discussion.  At that point the render callback would exist on the "Viewport" type rather than the Circuit type and most/all of the `RenderHelper`/`RenderOptions` would live outside `API.Circuit` too.